overlord/snapshotstate/snapshotstate.go

```diff
@@ -333,6 +333,19 @@ func Import(ctx context.Context, st *state.State, r io.Reader) (setID uint64, sn
 	}
 	snapNames, err = backendImport(ctx, setID, r)
 	if err != nil {
+		if dupErr, ok := err.(backend.DuplicatedSnapshotImportError); ok {
+			st.Lock()
+			defer st.Unlock()
+			// trying to import identical snapshot; instead return set ID of
+			// the existing one and reset its expiry time.
+			// XXX: at the moment expiry-time is the only attribute so we can
```

Comment:
Isn't the auto flag also something we store into the state?

Reply:
No, it was stored inside snapshot json, but in the state we only keep expiry-time for automatic snaphots.